File: CasaBackend/Casa.InterfaceAdapter/DTOs/ArduinoAutomationDto.cs
namespace CasaBackend.Casa.InterfaceAdapter.DTOs
{
    public class ArduinoAutomationDto
    {
        public int Id { get; set; }
        public int StartHour { get; set; }
        public int StartMinute { get; set; }
        public int EndHour { get; set; }
        public int EndMinute { get; set; }
        public byte Days { get; set; }
        public bool State { get; set; }
        public ICollection<AutomationDeviceDto> Devices { get; set; } = new List<AutomationDeviceDto>();
    }
}

File: CasaApp/src/application/dtos/ArduinoAutomationDto.ts
export interface AutomationDeviceDto {
  Id: number;
  State: boolean;
}

export class ArduinoAutomationDto {
  constructor(
    public readonly StartHour: number,
    public readonly StartMinute: number,
    public readonly EndHour: number,
    public readonly EndMinute: number,
    public readonly Days: number,
    public readonly State: boolean,
    public readonly Devices: AutomationDeviceDto[],
    public readonly Id: number,
  ) {
    this.Devices = Devices ?? [];
  }
}
